Additional update for UI

Jay-o-Way opened the pull request:
First step. Hope you find it helpful. Also for @SpiritReader and @Armin2208!
Much more to do later. I added a number of comments marked with **TODO** in the code.
Feel free to add changes here or suggest me to (try and) work on something.

![ADM winui](https://github.com/user-attachments/assets/85e95386-d005-4689-a76f-53917ef2712d)

Jay-o-Way committed:
initial update for UI

Jay-o-Way committed:
initial update for UI

Jay-o-Way committed:
small text improvs

Jay-o-Way committed:
Merge branch 'winui3' of https://github.com/Jay-o-Way/Windows-Auto-Night-Mode into winui3

Jay-o-Way commented:
Reference https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/issues/925

ChenYiLins commented on AutoDarkModeApp/App.xaml.cs:
This shouldn't be deleted because the WallpaperPicker has been enabled in the latest commit.

Jay-o-Way replied:
Hm, I didn't make this change "by hand", but I'll revert.

ChenYiLins commented on AutoDarkModeApp/Services/PageService.cs:
The same reason as above.

ChenYiLins commented on AutoDarkModeApp/Utils/Handlers/MessageHandler.cs:
> You've configured your code formatter to expand short hand get setters
> 
> ```cs
> public AdmConfig Config { get; private set; }
> ```
> 
> to
> 
> ```cs
> public AdmConfig Config
> {
>     get; private set;
> }
> ```
As @Spiritreader said, code styles should be consistent.

Jay-o-Way replied:
Wasn't aware of this, but I'll look into it.

ChenYiLins commented on AutoDarkModeApp/Strings/en-us/Resources.resw:
The change about the text is good. Some appropriate expressions will make users understand better. But the bigger question is whether the localized texts of other languages will need to be changed at the same time.

Jay-o-Way committed:
revert some CS files

unintentional changes

ChenYiLins commented on AutoDarkModeApp/Views/AboutPage.xaml:
I'm in great confusion now. ＞﹏＜ Should we wrap all of the control in SettingsCard? If not, which controls should not be wrapped in SettingsCard? I saw this design in [PowerToys](https://github.com/microsoft/PowerToys), but only HyperlinkButton. But in [WinUI Gallery](https://github.com/microsoft/WinUI-Gallery), all the controls are contained in SettingsCard. Which design scheme should we refer to? ≡(▔﹏▔)≡

Jay-o-Way replied:
I see what you mean. Good question. Maybe this should be a _discussion_ to collect feedback from different people. Looking at Windows Settings, many things (like hyperlinks) are still in a card or an expander...
Note that I wasn't completely finished with this PR ;)

ChenYiLins replied:
You mentioned Windows settings, this seems to be a good reference object. The HyperLinkButton at the bottom of the page may not be in the SettingsCard.

Jay-o-Way commented:
Looks like the whole WallPaperPicker wasn't in the files when I worked on this. Gotta add that too. Note that the clickable SettingsCard(s) on Personalization page doesn't do anything in Debug!

ChenYiLins commented:
> Looks like the whole WallPaperPicker wasn't in the files when I worked on this. Gotta add that too. Note that the clickable SettingsCard(s) on Personalization page doesn't do anything in Debug!

@Jay-o-Way What seems to be the mistake? Here is the code about `PersonalizationPage.xaml.cs`.
```c#
    private void PickWallpaperSettingsCard_Click(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
    {
        var navigation = App.GetService<INavigationService>();
        if (navigation?.Frame != null)
        {
            navigation.NavigateTo(typeof(WallpaperPickerViewModel).FullName!);
        }
    }
```
You could see it from [01f4ea](https://github.com/ChenYiLins/Windows-Auto-Night-Mode/commit/01f4ea68a249b35619d576903f75bc7cb11ededd).

ChenYiLins commented on AutoDarkModeApp/Views/PersonalizationPage.xaml.cs:
There is some code missing here: 
```c#
    private void PickWallpaperSettingsCard_Click(object sender, Microsoft.UI.Xaml.RoutedEventArgs e)
    {
        var navigation = App.GetService<INavigationService>();
        if (navigation?.Frame != null)
        {
            navigation.NavigateTo(typeof(WallpaperPickerViewModel).FullName!);
        }
    }
```

ChenYiLins commented on AutoDarkModeApp/Views/PersonalizationPage.xaml:
There is some code missing here: `Click="PickWallpaperSettingsCard_Click"/>`.

Jay-o-Way replied:
Thanks, found it!

ChenYiLins commented on AutoDarkModeApp/Views/TimePage.xaml:
>always show the TimePicker controls for light and dark (but invert bool on `IsWindowsNightLight`)

You put forward a brand-new idea, which seems to be very good. It's just that I'm not sure if this design method is intuitive, because time picker controls for light and dark are not always used in every mode.

Jay-o-Way replied:
Yes, it was just an idea. Reverted this in the mean time. Working on things now.

Jay-o-Way committed:
improvements

Jay-o-Way committed:
XAML TitleBar and update Navigation

- move 2 items to FooterMenuItems
- set properties for adavtive Nav
- remove theme and NavHeader related files

Jay-o-Way commented:
Updated TitleBar and NavigationView
![ADM title and nav](https://github.com/user-attachments/assets/f405c26d-e2d2-4eeb-b048-96d038d7b040)

ChenYiLins commented:
> Updated TitleBar and NavigationView更新了 TitleBar 和 NavigationView

@Jay-o-Way After testing, it looks very good now. (～￣▽￣)～. Do we need to raise the height of TitleBar?

Jay-o-Way commented:
> > Updated TitleBar and NavigationView更新了 TitleBar 和 NavigationView
> 
> @Jay-o-Way After testing, it looks very good now. (～￣▽￣)～. Do we need to raise the height of TitleBar?

Thanks! [edit: updated with force-push]
Height of titlebar is set automatically, depending if there are (interactive) elements in the Content part. There are none here.

Jay-o-Way commented:
> Do we need to raise the height of TitleBar?

Hm, maybe you are correct. The Back and Pane buttons _are_ interactive 🤔 I wonder why the control doesn't set this.

ChenYiLins commented:
> > Do we need to raise the height of TitleBar?
> 
> Hm, maybe you are correct. The Back and Pane buttons _are_ interactive 🤔 I wonder why the control doesn't set this.

I can't find a description of the height for the new TitleBar control. Do we still need to use the code that worked in the previous version? `App.MainWindow.AppWindow.TitleBar.PreferredHeightOption = TitleBarHeightOption.Tall;` But I always feel something is wrong.

Spiritreader commented:
This is really awesome!
I'm not super familiar with winui3 yet, but everything is already a lot more structured and sensible (compared to the old UI)
I'll have to take a closer look at this PR once I get the chance.

One thing that is very important to me is attribution!
I would like for you two (@Jay-o-Way , @ChenYiLins) to have proper credit on the About page that you worked on the winui3 rewrite.


Maybe an extra section here
![image](https://github.com/user-attachments/assets/1eb39bf2-4997-4e84-80bc-fb041547e4c5)

similar to special thanks with an attribution text, for example

"WinUI3 by @ChenYiLins & @Jay-o-Way"

What do you guys think?

Also pinging @Armin2208

Jay-o-Way commented:
> I can't find a description of the height for the new TitleBar control. Do we still need to use the code that worked in the previous version? `App.MainWindow.AppWindow.TitleBar.PreferredHeightOption = TitleBarHeightOption.Tall;` But I always feel something is wrong.

Code-behind should work,but I strongly recommend to set in XAML. Just try to add a property, if you type the first letters of `Height` it should be suggested. But in any case, low priority and quick fix.

Jay-o-Way commented:
Thanks @Spiritreader ! We're not at the target yet, but together we'll get a long way.

> Maybe an extra section here

In general, let's keep/start things short and simple. I think one section with "thanks" would be enough. When the time comes, online docs (including release notes) are good too.

ChenYiLins commented:
@Spiritreader Thank you very much! My idea is the same as @Jay-o-Way .o(￣€￣)o

Armin2208 commented:
Yep, they can be both added to the already existing special thanks to section. And next to that, we can now write why they are on the list.

Armin2208 commented:
Telegram is probably no one who sues people, but please use a telegram icon from the official asset package. Usually companies don't allow logo modifications. This is why we use a blue Telegram icon.
https://telegram.org/tour/screenshots/

ChenYiLins commented:
> Telegram is probably no one who sues people, but please use a telegram icon from the official asset package. Usually companies don't allow logo modifications. This is why we use a blue Telegram icon. https://telegram.org/tour/screenshots/

I will update the font file about icons. Similarly, do we need to add additional icons about Microsoft Office, PayPal?

ChenYiLins commented:
@Jay-o-Way Is it time to merge this PR? The work now seems to be fruitful.

Jay-o-Way commented:
> @Jay-o-Way Is it time to merge this PR? The work now seems to be fruitful.

Yes. 👍🏻 Let's push this to the ADM repository, so we can all use that branch to create new PRs against.

Armin2208 commented:
> I will update the font file about icons. Similarly, do we need to add additional icons about Microsoft Office, PayPal?

Kick out the Microsoft Office under "Apps". It's not necessary anymore. Yes, you need an icon for PayPal also for the donation page. It's already there in the current version and it looks nice.

Jay-o-Way commented:
> Kick out the Microsoft Office under "Apps". It's not necessary anymore.

I will add this to my open PR
